methods.py

```diff
@@ -1473,6 +1484,27 @@ def format_key_value(v):
         sys.exit()
 
 
+############################################################
+# FILE GENERATION & FORMATTING
+############################################################
+
+
+NodeOrPath = Union[str, "Node"]
+
+
+def node_to_path(value: NodeOrPath) -> str:
+    """
+    Converts a raw string / SCons Node to a pure string path, Regardless of what
+    type is passed in, the value is normalized & assigned posix separators.
+    """
+    if not isinstance(value, str):
+        if not hasattr(value, "get_abspath"):
+            raise TypeError(f'Expected type "str" or "Node"; was passed {type(value)}.')
+        value = value.get_abspath()
+
+    return os.path.normpath(str(value)).replace("\\", "/")
```

Comment:
Do we really need it and especially with absolute paths? There is native to SCons [GetBuildPath](https://scons.org/doc/production/HTML/scons-user.html#id1308) which does the same, gets string or Node and returns string representation.

 But it supports this operation even for arrays and allows variable expansion if we will need it. 

Though it isn't consistent in returning absolute paths